File: src/components/Home/WhyChoose.js
import React from "react";

const WhyChoose = () => {
  return (
    <section className="py-16 bg-gray-50">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="text-center mb-12">
          <h2 className="text-3xl font-bold font-poppins text-gray-900 mb-4">
            Pourquoi choisir EduNet ?
          </h2>
          <p className="text-gray-600 max-w-2xl mx-auto">
            Rejoignez une plateforme conçue pour maximiser votre apprentissage
          </p>
        </div>

        <div className="grid md:grid-cols-3 gap-8">
          <div className="text-center">
            <div className="bg-blue-100 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
              <i className="fas fa-award text-blue-600 text-2xl"></i>
            </div>
            <h3 className="font-semibold text-lg mb-2">Qualité</h3>
            <p className="text-gray-600">
              Cours créés par des experts reconnus dans leur domaine
            </p>
          </div>
          <div className="text-center">
            <div className="bg-blue-100 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
              <i className="fas fa-universal-access text-blue-600 text-2xl"></i>
            </div>
            <h3 className="font-semibold text-lg mb-2">Accessibilité</h3>
            <p className="text-gray-600">
              Apprenez à votre rythme, où que vous soyez
            </p>
          </div>
          <div className="text-center">
            <div className="bg-blue-100 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
              <i className="fas fa-chart-bar text-blue-600 text-2xl"></i>
            </div>
            <h3 className="font-semibold text-lg mb-2">Suivi</h3>
            <p className="text-gray-600">
              Suivez votre progression et obtenez des certificats
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

export default WhyChoose;

File: lib/components/home/why_choose.dart
// lib/components/home/why_choose.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:font_awesome_flutter/font_awesome_flutter.dart';

class WhyChoose extends StatelessWidget {
  const WhyChoose({super.key});

  @override
  Widget build(BuildContext context) {
    return Container(
      color: const Color(0xFFF3F4F6),
      padding: const EdgeInsets.symmetric(vertical: 64, horizontal: 16),
      child: Container(
        constraints: const BoxConstraints(maxWidth: 1280),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(
              'Pourquoi choisir EduNet ?',
              style: GoogleFonts.poppins(
                fontSize: 30,
                fontWeight: FontWeight.bold,
                color: const Color(0xFF111827),
              ),
            ),
            const SizedBox(height: 16),
            const Text(
              'Rejoignez une plateforme conçue pour maximiser votre apprentissage',
              style: TextStyle(
                fontSize: 16,
                color: Color(0xFF4B5563),
              ),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 32),
            LayoutBuilder(
              builder: (context, constraints) {
                bool isMobile = constraints.maxWidth < 768;

                if (isMobile) {
                  // Mobile: Stack vertically, no grid
                  return Column(
                    children: [
                      _buildCard(
                        icon: FontAwesomeIcons.award,
                        title: 'Qualité',
                        description: 'Cours créés par des experts reconnus',
                      ),
                      const SizedBox(height: 32),
                      _buildCard(
                        icon: FontAwesomeIcons.universalAccess,
                        title: 'Accessibilité',
                        description: 'Apprenez à votre rythme',
                      ),
                      const SizedBox(height: 32),
                      _buildCard(
                        icon: FontAwesomeIcons.chartBar,
                        title: 'Suivi',
                        description: 'Suivez votre progression',
                      ),
                    ],
                  );
                }

                // Desktop: Grid
                return Row(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    Expanded(
                      child: _buildCard(
                        icon: FontAwesomeIcons.award,
                        title: 'Qualité',
                        description: 'Cours créés par des experts reconnus dans leur domaine',
                      ),
                    ),
                    const SizedBox(width: 32),
                    Expanded(
                      child: _buildCard(
                        icon: FontAwesomeIcons.universalAccess,
                        title: 'Accessibilité',
                        description: 'Apprenez à votre rythme, où que vous soyez',
                      ),
                    ),
                    const SizedBox(width: 32),
                    Expanded(
                      child: _buildCard(
                        icon: FontAwesomeIcons.chartBar,
                        title: 'Suivi',
                        description: 'Suivez votre progression et obtenez des certificats',
                      ),
                    ),
                  ],
                );
              },
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildCard({
    required IconData icon,
    required String title,
    required String description,
  }) {
    return Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        Container(
          width: 64,
          height: 64,
          decoration: const BoxDecoration(
            color: Color(0xFFDBEAFE),
            shape: BoxShape.circle,
          ),
          child: Icon(icon, color: const Color(0xFF2563EB), size: 32),
        ),
        const SizedBox(height: 16),
        Text(
          title,
          style: const TextStyle(
            fontSize: 18,
            fontWeight: FontWeight.w600,
          ),
        ),
        const SizedBox(height: 8),
        Text(
          description,
          style: const TextStyle(
            color: Color(0xFF4B5563),
            fontSize: 14,
          ),
          textAlign: TextAlign.center,
        ),
      ],
    );
  }
}